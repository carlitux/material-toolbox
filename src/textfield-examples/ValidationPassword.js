// @flow
import * as React from 'react';

import Example from '../Example';
import TextField from '../material-toolbox/textfield/TextField';

const source = `import TextField from 'material-toolbox/textfield/TextField';

export default class ValidationPassword extends React.Component {
  state = {
    hasError: false,
    password: '',
  };

  handleChange = (evt) => {
    const { value } = evt.currentTarget;
    this.setState({
      password: value,
      hasError: value.length < 8,
    });
  };

  render() {
    return (
      <TextField
        type="password"
        label="Choose password"
        value={this.state.password}
        invalid={this.state.hasError}
        onChange={this.handleChange}
        helperText="Must be at least 8 characters long"
        validationHelperText={this.state.hasError}
        persistentHelperText
        required
      />
    );
  }
}
`;

type Props = {};
type State = {
  hasError: boolean,
  password: string,
};

export default class ValidationPassword extends React.Component<Props, State> {
  state = {
    hasError: false,
    password: '',
  };

  handleChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
    const { value } = evt.currentTarget;
    this.setState({
      password: value,
      hasError: value.length < 8,
    });
  };

  render() {
    return (
      <div>
        <Example title="Password field with validation" source={source}>
          <TextField
            type="password"
            label="Choose password"
            value={this.state.password}
            invalid={this.state.hasError}
            helperText="Must be at least 8 characters long"
            onChange={this.handleChange}
            validationHelperText={this.state.hasError}
            persistentHelperText
            required
          />
        </Example>
      </div>
    );
  }
}
