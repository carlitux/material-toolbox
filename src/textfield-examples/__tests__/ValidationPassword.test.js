// @flow
import React from 'react';
import { shallow } from 'enzyme';
import ValidationPassword from '../ValidationPassword';

describe('example::ValidationPassword', () => {
  it('Should match snapshoot', () => {
    const component = shallow(<ValidationPassword />);
    expect(component).toMatchSnapshot();
  });
});
