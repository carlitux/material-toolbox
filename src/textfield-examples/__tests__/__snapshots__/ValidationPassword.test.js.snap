// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example::ValidationPassword Should match snapshoot 1`] = `
<div>
  <Example
    source="import TextField from 'material-toolbox/textfield/TextField';

export default class ValidationPassword extends React.Component {
  state = {
    hasError: false,
    password: '',
  };

  handleChange = (evt) => {
    const { value } = evt.currentTarget;
    this.setState({
      password: value,
      hasError: value.length < 8,
    });
  };

  render() {
    return (
      <TextField
        type=\\"password\\"
        label=\\"Choose password\\"
        value={this.state.password}
        invalid={this.state.hasError}
        onChange={this.handleChange}
        helperText=\\"Must be at least 8 characters long\\"
        validationHelperText={this.state.hasError}
        persistentHelperText
        required
      />
    );
  }
}
"
    title="Password field with validation"
  >
    <TextField
      box={false}
      cssOnly={false}
      dense={false}
      disabled={false}
      fullWidth={false}
      helperText="Must be at least 8 characters long"
      invalid={false}
      label="Choose password"
      multiline={false}
      onChange={[Function]}
      outlined={false}
      persistentHelperText={true}
      required={true}
      type="password"
      validationHelperText={false}
      value=""
    />
  </Example>
</div>
`;
